keps/sig-storage/1710-selinux-relabeling/README.md

```diff
@@ -111,153 +132,247 @@ Following conditions must be met:
+// ...
+	// seLinuxChangePolicy defines how the container's SELinux label is applied to all volumes used by the Pod.
+	// It has no effect on nodes that do not support SELinux or when the volume does not support SELinux.
+    // Valid values are "MountOption" and "Recursive". If not specified, "MountOption" is used.
+	// It affects only in-tree iSCSI and FibreChannel volumes, and CSI volumes that announce "SELinuxMount: true" in their CSIDriver instance.
+	// It affects only Pods that have SELinux label set, either in PodSecurityContext or in SecurityContext of all containers.
+    // Note that this field cannot be set when spec.os.name is windows.
+    // +optional
+    SELinuxChangePolicy *PodSELinuxChangePolicy
+}
+
+// PodSELinuxChangePolicy defines how the container's SELinux label is applied to all volumes used by the Pod.
+type PodSELinuxChangePolicy string
+
+const (
+	// Recursive relabeling of all Pod volumes by the container runtime.
+	// This may be slow for large volumes, but allows mixing privileged and unprivileged Pods sharing the same volume on the same node.
+    SELinuxChangePolicyRecursive PodSELinuxChangePolicy = "Recursive"
+	// MountOption mounts all Pod volumes with `-o context` mount option.
+	// This requires all Pods that share the same volume to use the same SELinux label.
+	// It is not possible to share the same volume among privileged and unprivileged Pods.
+    SELinuxChangePolicyMountOption PodSELinuxChangePolicy = "MountOption"
+)
+```
+
+#### Conflicts with other Pods
+
+It is Pod (Deployment, StatefulSet, DaemonSet) author responsibility to set the `SELinuxChangePolicy` and SELinux labels on Pods correctly.
+For example, these cases can happen if they are careless.
+All these cases assume that the volume used in the examples does support mount with `-o context` and all Pods run on the same node, and they access the same volume.
+
+**Mixing Pods with different SELinux labels**: In both cases we can see that only one of the pods can run and access the volume, but they differ in which Pod it is.
+* Pod A with `SELinuxChangePolicy: Recursive` and label `c1,c2` runs, Pod B with `SELinuxChangePolicy: Recursive` with label `c8,c9` is about to start.
+  * This is the behavior before this KEP. All files on the volume were relabeled to `c1,c2` when A started. When Pod B starts, the container runtime relabels all files on the volume to `c8,c9` and Pod A loses access to data on the volume (i.e. it will get `EPERM` to all OS calls).
+* Pod A with `SELinuxChangePolicy: MountOption` and label `c1,c2` runs, Pod B with `SELinuxChangePolicy: MountOption` with label `c8,c9` is about to start.
+  * Kubelet mounted the volume with `-o context=c1,c2` for Pod A. Pod B wants the same volume mounted with `-o context=c8,c9`. Pod B will stay in `ContainerCreating` state until Pod A finishes and kubelet unmounts the volume, so it can be re-mounted for Pod B. Kubelet will provide an event that describes why Pod B is not running.
```

Comment:
I think if pod A and B, both use `MountOption` policy and they use different selinux contexts but get scheduled to different nodes, they both will be able to start. Previously - when using `Recursive` one pod would lock out the other.

Reply:
Good point, added a new "Multiple-node conflicts" section.